libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -55,30 +55,38 @@ export class TopicRepository extends BaseRepository<TopicDBModel, TopicEntity, E
   async filterTopics(
     query: FilterQuery<TopicDBModel>,
     pagination: { limit: number; skip: number }
-  ): Promise<TopicEntity & { subscribers: ExternalSubscriberId[] }[]> {
-    const parsedQuery = { ...query };
+  ): Promise<{ data: TopicEntity & { subscribers: ExternalSubscriberId[] }[]; totalCount: number }> {
+    const { subscriberId, ...parsedQuery } = query;
     if (query._id) {
       parsedQuery._id = this.convertStringToObjectId(query._id);
     }
 
     parsedQuery._environmentId = this.convertStringToObjectId(query._environmentId);
     parsedQuery._organizationId = this.convertStringToObjectId(query._organizationId);
-
-    const data = await this.aggregate([
-      {
-        $match: parsedQuery,
-      },
-      lookup,
-      topicWithSubscribersProjection,
-      {
-        $skip: pagination.skip,
-      },
-      {
-        $limit: pagination.limit,
+    const subscriberFilter = subscriberId && {
+      $match: {
+        $expr: {
+          $in: [subscriberId, '$topicSubscribers.externalSubscriberId'],
+        },
       },
+    };
+    const commonAggQuery = [{ $match: parsedQuery }, lookup, ...(subscriberFilter ? [subscriberFilter] : [])];
+
+    const [data, totalCount] = await Promise.all([
+      this.aggregate([
```

Comment:
Although i like the optimization here with one mongo query, maybe we can keep the previous implementation but add `Promise.alll` to the two queries? my main concerns are that not filterTopics is responsible for two actions (get data and get totalCount) and the additional complexity.
We most definitely can back to it once we notice that we have a high load in here and we need to optimize more. wdyt?